distributed-prover/src/circuits/evm.rs

```diff
@@ -0,0 +1,216 @@
+use ark_std::{end_timer, start_timer};
+use bus_mapping::{circuit_input_builder::FixedCParams, mock::BlockData};
+use eth_types::geth_types::GethData;
+use halo2_proofs::{
+    halo2curves::bn256::{Bn256, Fr, G1Affine},
+    plonk::{
+        create_proof as create_proof_local, distributed_prover::{
+            config:: WorkloadConfig,
+            prover::create_proof as create_proof_distributed,
+        },
+        keygen_pk, keygen_vk, verify_proof,
+        Circuit,
+    },
+    poly::{
+        commitment::ParamsProver,
+        kzg::{
+            commitment::{KZGCommitmentScheme, ParamsKZG},
+            multiopen::{ProverSHPLONK, VerifierSHPLONK},
+            strategy::SingleStrategy,
+        },
+    },
+    transcript::{
+        Blake2bRead, Blake2bWrite, Challenge255, TranscriptReadBuffer, TranscriptWriterBuffer,
+    },
+};
+use mock::TestContext;
+use rand::SeedableRng;
+use rand_xorshift::XorShiftRng;
+use zkevm_circuits::{
+    evm_circuit::{EvmCircuit, witness::block_convert, TestEvmCircuit},
+    util::SubCircuit
+};
+
+use crate::util::artifacts::*;
+use crate::util::constants::RNG_SEED;
+
+pub(crate) const CIRCUIT_NAME: &str = "evm";
+const CIRCUIT_DEGREE: u32 = 18;
+
+// What is the <Fr> notation?
+// What is the pub(crate) notation?
+pub(crate) fn circuit() -> EvmCircuit<Fr> { 
+    let timer = start_timer!(|| "Create circuit");  // What is the "||" Notation?
+
+    let empty_data: GethData = TestContext::<0, 0>::new(None, |_| {}, |_, _| {}, |b, _| b)
+        .unwrap()
+        .into();
+
+    let mut builder =
+        BlockData::new_from_geth_data_with_params(empty_data.clone(), FixedCParams::default())
+            .new_circuit_input_builder();
+
+    builder
+        .handle_block(&empty_data.eth_block, &empty_data.geth_traces)
+        .unwrap();
+
+    let block = block_convert(&builder).unwrap();
+
+    let circuit = TestEvmCircuit::<Fr>::new(block);
+
+    end_timer!(timer);
+
+    circuit
+}
+
+pub(crate) fn setup() {
+    let circuit = circuit();
+
+    // Setup params
+    let timer = start_timer!(|| "Set up params");
+    let mut rng = XorShiftRng::from_seed(RNG_SEED);
+    let general_params = ParamsKZG::<Bn256>::setup(CIRCUIT_DEGREE, &mut rng);
+    let verifier_params = general_params.verifier_params().clone();     // What's the purpose of cloning here?
```

Comment:
In Rust, assignment is move by default, so cloning is needed in some places.

However, good catch here. I don't think cloning is needed.